File: app/src/main/java/com/mesada/shop/adapter/CartAdapter.java
package com.mesada.shop.adapter;

import android.graphics.Color;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.mesada.shop.databinding.CartRowBinding;
import com.mesada.shop.model.Cart;

import static androidx.databinding.library.baseAdapters.BR.cartItem;
import static androidx.databinding.library.baseAdapters.BR.categoryInterface;

public class CartAdapter extends ListAdapter<Cart, CartAdapter.CartViewHolder> {

    private CartInterface cartInterface;
    public CartAdapter(CartInterface cartInterface) {
        super(Cart.itemCallback);
        this.cartInterface = cartInterface;
    }

    @NonNull
    @Override
    public CartViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        CartRowBinding cartRowBinding = CartRowBinding.inflate(layoutInflater, parent, false);
        return new CartViewHolder(cartRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull CartViewHolder holder, int position) {
        holder.cartRowBinding.setCartItem(getItem(position));
        holder.cartRowBinding.cartBackground.setBackgroundColor(Color.parseColor(getItem(position).getProduct().getBgColor()));
        holder.cartRowBinding.executePendingBindings();
    }

    class CartViewHolder extends RecyclerView.ViewHolder {

        CartRowBinding cartRowBinding;
        public CartViewHolder(@NonNull CartRowBinding cartRowBinding) {
            super(cartRowBinding.getRoot());
            this.cartRowBinding = cartRowBinding;

            cartRowBinding.deleteProductButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    cartInterface.deleteItem(getItem(getAdapterPosition()));
                }
            });

        }
    }

    public interface CartInterface {
        void deleteItem(Cart cartItem);
        void changeQuantity(Cart cartItem, int quantity);
    }
}

File: app/src/main/java/com/mesada/shop/adapter/CartAdapter.kt
package com.mesada.shop.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.mesada.shop.adapter.CartAdapter.CartViewHolder
import com.mesada.shop.databinding.CartRowBinding
import com.mesada.shop.model.Cart

class CartAdapter(private val cartInterface: CartInterface) : ListAdapter<Cart, CartViewHolder>(Cart.itemCallback) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CartViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val cartRowBinding = CartRowBinding.inflate(layoutInflater, parent, false)
        return CartViewHolder(cartRowBinding)
    }

    override fun onBindViewHolder(holder: CartViewHolder, position: Int) {
        holder.cartRowBinding.cartItem = getItem(position)
        holder.cartRowBinding.cartBackground.setBackgroundResource(getItem(position)!!.product!!.imageResourceBg)
        holder.cartRowBinding.executePendingBindings()
    }

    inner class CartViewHolder(var cartRowBinding: CartRowBinding) : RecyclerView.ViewHolder(cartRowBinding.root) {

        init {
            cartRowBinding.deleteProductButton.setOnClickListener { cartInterface.deleteItem(getItem(adapterPosition)) }
        }
    }

    interface CartInterface {
        fun deleteItem(cartItem: Cart?)
        fun changeQuantity(cartItem: Cart?, quantity: Int)
    }

}